### PR TITLE
Add auth to api kwargs

### PR DESCRIPTION
**Pull request by mdellweg**
All requests auth class compatible auth mechanisms can now be passed to the PulpContext.

[noissue]

Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

**Comment by mdellweg** [comment]
I plan to redraft this to use the openapi3 docs helping to decide on the authentication mechanism.

File: pulpcore/cli/ansible/content.py [hunk 1]
```diff
@@ -47,7 +47,6 @@ def _fields_callback(ctx: click.Context, param: click.Parameter, value: Any) ->
     if value:
         click.echo(_("Option {name} is deprecated.").format(name=param.opts[0]), err=True)
         value = tuple(value.split(","))
-        print(value)
```
Comment (ggainey) [inline pulpcore/cli/ansible/content.py:50]: oops :)

File: pulpcore/cli/common/generic.py [hunk 2]
```diff
@@ -105,9 +110,6 @@ def __init__(
     def echo(self, message: str, nl: bool = True, err: bool = False) -> None:
         click.echo(message, nl=nl, err=err)
 
-    def prompt(self, text: str, hide_input: bool = False) -> t.Any:
```
Comment (ggainey) [inline pulpcore/cli/common/generic.py:108]: I assume this was a never-used (and kind of pointless) method?

**Review by ggainey — APPROVED** [review]

File: pulpcore/cli/ansible/content.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/ansible/content.py:50]: indeed.

File: pulpcore/cli/common/generic.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/cli/common/generic.py:108]: Let's say this change pushes the ask-me-for-a-password one abstraction level down where we can use `click.promt` directly.

**Review by ggainey — APPROVED** [review]

**Pull request merged**